### PR TITLE
Minor code restructurings after merge of `gardener/gardener-resource-manager`

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement
/merge squash

**What this PR does / why we need it**:
After #4757, there is quite some potential to improve the code structure and remove duplicated functions.
This PR covers a couple of aspects.

Eventually, I think we should move `extensions/pkg/...` into `pkg/extensions/...` and `extensions/test` into `test/...` for symmetry reasons with other packages.

**Release note**:

```breaking developer
Some functions were moved from `extensions/...` to other packages. You might need to adapt your import paths.
```

### CODE REVIEW

**Comment by timebertt** [comment]
> Eventually, I think we should move `extensions/pkg/...` into `pkg/extensions/...` and `extensions/test` into `test/...` for symmetry reasons with other packages.

+1

We have some package dependencies from `pkg` to `extensions` which we originally wanted to avoid. I think, these should actually be inverted (even if we move `extensions` into `pkg`):
```
$ grep -R '"github.com/gardener/gardener/extensions' --exclude-dir=vendor pkg
pkg/operation/botanist/component/extensions/operatingsystemconfig/utils/filecontentinlinecodec.go:	"github.com/gardener/gardener/extensions/pkg/controller/operatingsystemconfig/oscommon/cloudinit"
pkg/operation/botanist/component/extensions/operatingsystemconfig/utils/filecontentinlinecodec_test.go:	"github.com/gardener/gardener/extensions/pkg/controller/operatingsystemconfig/oscommon/cloudinit"
pkg/operation/botanist/etcd.go:	"github.com/gardener/gardener/extensions/pkg/controller/backupentry/genericactuator"
pkg/apis/extensions/validation/operatingsystemconfig.go:	"github.com/gardener/gardener/extensions/pkg/controller/operatingsystemconfig/oscommon/cloudinit"
pkg/gardenlet/controller/bastion/bastion_reconciler.go:	extensionscontroller "github.com/gardener/gardener/extensions/pkg/controller"
pkg/gardenlet/controller/bastion/bastion_reconciler.go:	ctrlerror "github.com/gardener/gardener/extensions/pkg/controller/error"
```

File: pkg/controllerutils/predicate/has_name.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
```
Comment (timebertt) [inline pkg/controllerutils/predicate/has_name.go:1]: Why so many small files in `pkg/controllerutils/predicate`?

File: extensions/pkg/predicate/predicate.go [hunk 2]
```diff
@@ -18,12 +18,12 @@ import (
 	"errors"
 
 	extensionscontroller "github.com/gardener/gardener/extensions/pkg/controller"
-	extensionsinject "github.com/gardener/gardener/extensions/pkg/inject"
 	gardencore "github.com/gardener/gardener/pkg/api/core"
 	"github.com/gardener/gardener/pkg/api/extensions"
 	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
 	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
 	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	injectutils "github.com/gardener/gardener/pkg/controllerutils/inject"
```
Comment (timebertt) [inline extensions/pkg/predicate/predicate.go]: Actually, I think this package should rather be eliminated than moved. We don't use it consequently, and the injection is so simple, that there's no need for helpers – they just cause more confusion than some simple and obvious code.

**Review by timebertt — COMMENTED** [review]
Thanks a lot for the cleanup!
Looks quite good, only some minor comments from my side.

File: extensions/pkg/predicate/predicate.go (hunk 2 above)
Comment (rfranzke) [reply extensions/pkg/predicate/predicate.go]: Good point, I removed it.

File: pkg/controllerutils/predicate/has_name.go (hunk 1 above)
Comment (rfranzke) [reply pkg/controllerutils/predicate/has_name.go:1]: Why not? 😄 I think each predicate is independent of the others, so for a better overview I personally prefer to have dedicated files in such cases.

**Review by timebertt — APPROVED** [review]
Nice, thanks!
/lgtm

**Pull request merged**